test_scripts/Polices/App_Permissions/ATF_HP_Order_Of_Request_Response_Notification_On_Registering_App.lua

```diff
@@ -0,0 +1,143 @@
+---------------------------------------------------------------------------------------------
+-- Description:
+-- Application with appID intends to be registered on SDL, device with deviceID the application is running on is connected to HU.
+-- 1. Used preconditions:
+-- Restart SDL and clean all data to set SDL in first life cycle state.
+-- 2. Performed steps
+-- Register new application.
+--
+-- Requirement summary:
+-- The order of requests/responses/notifications during application registering must be the following:
+-- 1. app->SDL: RegisterAppInterface (policy_appID, parameters)
+-- 2. SDL->HMI: OnAppRegistered (hmi_appID, params)
+-- 3. SDL->app: RegisterAppInterface_response (<applicable resultCode>, success:true)
+-- 4. SDL->app: OnHMIStatus(hmiLevel,audioStreamingState, systemContext)
+-- 5. SDL->app: OnPermissionsChange(params)
+--
+-- Expected result:
+-- 1. On performing all checks for successful registering SDL notifies HMI about registering before sending a response to mobile application:
+-- SDL->HMI: OnAppRegistered(hmi_appID)
+-- 2. SDL->appID: (<applicable resultCode>, success:true): RegisterAppInterface()
+-- 3. On registering the application, HMIStatus parameters are assinged to the application:
+-- SDL->app: OnHMIStatus(hmiLevel,audioStreamingState, systemContext)
+-- 4. SDL assigns the appropriate policies and notifies application:
+-- SDL->app: OnPermissionsChange (params) - as specified in "pre_DataConsent" section.
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+--ToDo: shall be removed when issue: "ATF does not stop HB timers by closing session and connection" is fixed
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require ('user_modules/shared_testcases/commonSteps')
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+local mobile_session = require('mobile_session')
+
+--[[ Local Functions ]]
```

Comment:
@OVikhrov,  looks like there is no need in separate --[[ Local Functions ]] block since there are no functions except related to Preconditions, so function Precondition could be moved to block --[[ Preconditions ]] itself.